source/elements/oneMKL/source/domains/blas/syrk_batch.rst

```diff
@@ -0,0 +1,530 @@
+                              transpose *trans,
+                              std::int64_t *n,
+                              std::int64_t *k,
+                              T *alpha,
+                              const T **a,
+                              std::int64_t *lda,
+                              T *beta,
+                              T **c,
+                              std::int64_t *ldc,
+                              std::int64_t group_count,
+                              std::int64_t *group_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   queue
+      The queue where the routine should be executed.
+
+   upper_lower
+      Specifies whether data in ``C`` is stored in its upper or lower triangle.
+      For more details, see :ref:`onemkl_datatypes`.
+
+   trans
+      Array of ``group_count`` ``onemkl::transpose`` values. ``trans[i]`` specifies the form of op(``A``) used in
+      the rank-k update in group ``i``. See :ref:`onemkl_datatypes` for more details.
+
+   n
+      Array of ``group_count`` integers. ``n[i]`` specifies the
+      number of rows and columns of ``C`` for every matrix in group ``i``. All entries must be at least zero.
+
+   k
+      Array of ``group_count`` integers. ``k[i]`` specifies the
+      number of columns of op(``A``) for every matrix in group ``i``. All entries must be at
+      least zero.
+
+   alpha
+      Array of ``group_count`` scalar elements. ``alpha[i]`` specifies the scaling factor for every rank-k update in group ``i``.
+
+   a
+      Array of pointers to input matrices ``A`` with size ``total_batch_count``. 
+      
+      See :ref:`matrix-storage` for more details.
+
+   lda
+      Array of ``group_count`` integers. ``lda[i]`` specifies the
+      leading dimension of ``A`` for every matrix in group ``i``. All
+      entries must be positive.
+
+      .. list-table::
+         :header-rows: 1
+
+         * -
+           - ``A`` not transposed
+           - ``A`` transposed
+         * - Column major
+           - ``lda[i]`` must be at least ``m[i]``.
+           - ``lda[i]`` must be at least ``k[i]``.
+         * - Row major
+           - ``lda[i]`` must be at least ``k[i]``.
+           - ``lda[i]`` must be at least ``m[i]``.
+             
+   beta
+      Array of ``group_count`` scalar elements. ``beta[i]`` specifies the scaling factor for matrix ``C`` 
+      for every matrix in group ``i``.
+
+   c
+      Array of pointers to input/output matrices ``C`` with size ``total_batch_count``. 
+      
+      See :ref:`matrix-storage` for more details.
+
+   ldc
+      Array of ``group_count`` integers. ``ldc[i]`` specifies the
+      leading dimension of ``C`` for every matrix in group ``i``.  All
+      entries must be positive and ``ldc[i]`` must be at least
+      ``m[i]`` if column major layout is used to store matrices or at
```

Comment:
m[i]? C is n by n. So ldc[i] must be as least n[i]

Reply:
adjusted